src/minio/core/policy_creator.py

```diff
@@ -222,6 +224,68 @@ def _generate_policy_name(self) -> str:
         except Exception as e:
             raise PolicyOperationError(f"Generated policy name is invalid: {e}") from e
 
+    def _add_path_access_via_builder(
+        self, bucket_name: str, path: str, permission_level: PolicyPermissionLevel
+    ) -> None:
+        """
+        Add path access using PolicyBuilder and update internal sections.
+
+        Args:
+            path: The S3 path to grant access to
+            permission_level: The level of access to grant
+        """
+        # Get current policy state
+        current_policy = self._get_current_policy()
+
+        # Use PolicyBuilder to add the path access
+        builder = PolicyBuilder(current_policy, bucket_name)
+        updated_policy = builder.add_path_access(
+            path, permission_level, new_policy=True
+        ).build()
+
+        # Clear current sections and rebuild from updated policy
+        self._rebuild_sections_from_policy(updated_policy)
+
+    def _rebuild_sections_from_policy(self, policy: PolicyModel) -> None:
+        """
+        Rebuild internal sections from a policy model.
+
+        Args:
+            policy: Policy model to extract sections from
+        """
+        # Clear existing sections
+        for section_type in self._sections:
+            self._sections[section_type] = []
+
+        # Categorize statements back into sections
+        for stmt in policy.policy_document.statement:
+            if PolicyAction.LIST_ALL_MY_BUCKETS in (
+                stmt.action if isinstance(stmt.action, list) else [stmt.action]
+            ):
+                self._sections[PolicySectionType.GLOBAL_PERMISSIONS].append(stmt)
+            elif PolicyAction.GET_BUCKET_LOCATION in (
+                stmt.action if isinstance(stmt.action, list) else [stmt.action]
+            ):
+                self._sections[PolicySectionType.GLOBAL_PERMISSIONS].append(stmt)
+            elif PolicyAction.LIST_BUCKET in (
+                stmt.action if isinstance(stmt.action, list) else [stmt.action]
+            ):
+                self._sections[PolicySectionType.BUCKET_ACCESS].append(stmt)
+            elif PolicyAction.GET_OBJECT in (
+                stmt.action if isinstance(stmt.action, list) else [stmt.action]
+            ):
+                self._sections[PolicySectionType.READ_PERMISSIONS].append(stmt)
+            elif PolicyAction.PUT_OBJECT in (
+                stmt.action if isinstance(stmt.action, list) else [stmt.action]
+            ):
+                self._sections[PolicySectionType.WRITE_PERMISSIONS].append(stmt)
+            elif PolicyAction.DELETE_OBJECT in (
+                stmt.action if isinstance(stmt.action, list) else [stmt.action]
+            ):
+                self._sections[PolicySectionType.DELETE_PERMISSIONS].append(stmt)
+            else:
+                raise PolicyOperationError(f"Unsupported policy action: {stmt.action}")
```

Comment:
```suggestion
            for pa in (stmt.action if isinstance(stmt.action, list) else [stmt.action]);
                   self._sections[_POLICY_ACTION_TO_POLICY_SECTiON[pa]].append(stmt)
```

_POLICY_ACTION_TO_POLICY_SECTiON is a module attribute map